File: rust/src/model/domain/customer.rs
#[derive(Debug, Clone, Copy)]
pub struct Customer(pub u32);

impl Customer {
    pub fn id(&self) -> String {
        format!("CUST-{:05}", self.0)
    }

    pub fn name(&self) -> String {
        format!("Customer {:05}", self.0)
    }
}

File: golang/model/domain/customer.go
package domain

import (
	"fmt"
)

type Customer struct {
	no int
}

func NewCustomer(no int) Customer {
	return Customer{no}
}

func (u Customer) Id() string {
	return fmt.Sprintf("CUST-%05d", u.no)
}

func (u Customer) Name() string {
	return fmt.Sprintf("Customer %05d", u.no)
}

func (u Customer) No() int {
	return u.no
}
